src/ert/config/design_matrix.py

```diff
@@ -50,3 +72,134 @@ def from_config_list(cls, config_list: List[str]) -> "DesignMatrix":
+            raise ValueError(f"Design matrix not valid, error: {err!s}") from err
+
+        # Todo: Check for invalid realizations, drop them maybe?
+        # This should probably handle/(fill in) missing values in design_matrix_sheet as well? Or maybe not.
+        defaults = DesignMatrix._read_defaultssheet(
+            self.xls_filename, self.default_sheet
+        )
+        for k, v in defaults.items():
+            if k not in design_matrix_df.columns:
+                design_matrix_df[k] = v
+
+        # ignoring errors here is deprecated in pandas, should find another solution
+        # design_matrix_sheet = design_matrix_sheet.apply(pd.to_numeric, errors="ignore")
+        parameter_configuration: dict[str, ParameterConfig] = {}
+        transform_function_definitions: list[TransformFunctionDefinition] = []
+        for parameter in design_matrix_df.columns:
+            transform_function_definitions.append(
+                TransformFunctionDefinition(
+                    name=parameter,
+                    param_name="RAW",
+                    values=[],
+                )
+            )
+        parameter_configuration[DESIGN_MATRIX_GROUP] = GenKwConfig(
+            name=DESIGN_MATRIX_GROUP,
+            forward_init=False,
+            template_file=None,
+            output_file=None,
+            transform_function_definitions=transform_function_definitions,
+            update=False,
+        )
+
+        design_matrix_df.columns = pd.MultiIndex.from_product(
+            [[DESIGN_MATRIX_GROUP], design_matrix_df.columns]
+        )
+        reals = design_matrix_df.index.tolist()
+        self.num_realizations = len(reals)
+        self.active_realizations = [x in reals for x in range(max(reals))]
+
+        self.design_matrix_df = design_matrix_df
+        self.parameter_configuration = parameter_configuration
+
+    @staticmethod
+    def _read_excel(
+        file_name: Path | str,
+        sheet_name: str,
+        usecols: int | list[int] | None = None,
+        header: int | None = 0,
+    ) -> pd.DataFrame:
+        """
+        Make dataframe from excel file
+        :return: Dataframe
+        :raises: OsError if file not found
+        :raises: ValueError if file not loaded correctly
+        """
+        dframe: pd.DataFrame = pd.read_excel(
+            file_name, sheet_name, usecols=usecols, header=header, dtype={"REAL": int}
+        )
+        return dframe.dropna(axis=1, how="all")
+
+    def _validate_design_matrix_header(design_matrix: pd.DataFrame) -> None:
+        """
+        Validate header in user inputted design matrix
+        :raises: ValueError if design matrix contains empty headers
+        """
+        if design_matrix.empty:
+            return
+        try:
+            unnamed = design_matrix.loc[
+                :, design_matrix.columns.str.contains("^Unnamed")
+            ]
+        except ValueError as err:
+            # We catch because int/floats as column headers
```

Comment:
Don't understand this comment tbh

Reply:
As discussed I'd change the comment:
# this catches two errors in the header
# 1 - Empty column name, will produce Unnamed str
# 2 - Pure number, will produce Nan, which raises ValueError when used in df.loc